cocotb/run.py

```diff
@@ -0,0 +1,929 @@
+            if os.path.isdir(absworkdir):
+                self.work_dir = absworkdir
+
+        if python_search is None:
+            python_search = []
+
+        self.python_search = python_search
+
+        self.toplevel = toplevel
+        self.toplevel_lang = toplevel_lang
+
+        if verilog_sources is None:
+            verilog_sources = []
+
+        self.verilog_sources = self.get_abs_paths(verilog_sources)
+
+        if vhdl_sources is None:
+            vhdl_sources = []
+
+        self.vhdl_sources = self.get_abs_paths(vhdl_sources)
+
+        if includes is None:
+            includes = []
+
+        self.includes = self.get_abs_paths(includes)
+
+        if defines is None:
+            defines = []
+
+        self.defines = defines
+
+        if parameters is None:
+            parameters = {}
+
+        self.parameters = parameters
+
+        if compile_args is None:
+            compile_args = []
+
+        if extra_args is None:
+            extra_args = []
+
+        self.compile_args = compile_args + extra_args
+
+        if sim_args is None:
+            sim_args = []
+
+        self.sim_args = sim_args + extra_args
+
+        if plus_args is None:
+            plus_args = []
+
+        self.plus_args = plus_args
+        self.force_compile = force_compile
+        self.compile_only = compile_only
+
+        # by copy since we modify
+        self.env = dict(extra_env) if extra_env is not None else {}
+
+        if testcase is not None:
+            self.env["TESTCASE"] = testcase
+
+        if seed is not None:
+            self.env["RANDOM_SEED"] = str(seed)
+
+        if waves is None:
+            self.waves = bool(int(os.getenv("WAVES", 0)))
+        else:
+            self.waves = bool(waves)
+
+        self.gui = gui
+
+        # Catch SIGINT and SIGTERM
+        self.old_sigint_h = signal.getsignal(signal.SIGINT)
+        self.old_sigterm_h = signal.getsignal(signal.SIGTERM)
+
+        # works only if main thread
+        if threading.current_thread() is threading.main_thread():
+            signal.signal(signal.SIGINT, self.exit_gracefully)
+            signal.signal(signal.SIGTERM, self.exit_gracefully)
+
+        self.process = None
+
+    def set_env(self):
+
+        for e in os.environ:
+            self.env[e] = os.environ[e]
+
+        self.env["LIBPYTHON_LOC"] = find_libpython.find_libpython()
+
+        self.env["PATH"] += os.pathsep + self.lib_dir
+
+        self.env["PYTHONPATH"] = os.pathsep.join(sys.path)
+        for path in self.python_search:
+            self.env["PYTHONPATH"] += os.pathsep + path
+
+        self.env["PYTHONHOME"] = get_config_var("prefix")
+
+        self.env["TOPLEVEL"] = self.toplevel
+        self.env["MODULE"] = self.module
+
+        if not os.path.exists(self.sim_dir):
+            os.makedirs(self.sim_dir)
+
+    def build_command(self):
+        raise NotImplementedError()
+
+    def run(self):
+
+        sys.tracebacklimit = 0  # remove not needed traceback from assert
```

Comment:
This is probably a bad idea - if you want to exit without a traceback, use `raise SystemExit(message)`